src/main/java/sg/ncl/DataController.java

```diff
@@ -60,14 +72,19 @@ public String data(Model model, HttpSession session) {
         return "data";
     }
 
-    @RequestMapping(value={"/contribute", "/contribute/{id}"}, method= RequestMethod.GET)
-    public String contributeData(Model model, @PathVariable Optional<String> id) throws Exception {
+    @RequestMapping(value={"/contribute", "/contribute/{id}"}, method=RequestMethod.GET)
+    public String contributeData(Model model, @PathVariable Optional<String> id, HttpSession session, RedirectAttributes redirectAttributes) throws Exception {
         if (id.isPresent()) {
             HttpEntity<String> request = createHttpEntityHeaderOnly();
             ResponseEntity response = restTemplate.exchange(properties.getDataset(id.get()), HttpMethod.GET, request, String.class);
             String dataResponseBody = response.getBody().toString();
             JSONObject dataInfoObject = new JSONObject(dataResponseBody);
             Dataset dataset = extractDataInfo(dataInfoObject.toString());
+            if (!dataset.getContributorId().equals(session.getAttribute("id").toString())) {
+                log.warn(EDIT_DISALLOWED);
+                redirectAttributes.addFlashAttribute(MESSAGE_ATTRIBUTE, EDIT_DISALLOWED);
+                return "redirect:/data";
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Define a constant instead of duplicating this literal "redirect:/data" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1192)